fluss-server/src/main/java/org/apache/fluss/server/kv/snapshot/SharedKvFileRegistry.java

```diff
@@ -66,6 +66,22 @@ public SharedKvFileRegistry(Executor asyncDisposalExecutor) {
         this.open = true;
     }
 
+    public long getFileSize() {
+        synchronized (registeredKvEntries) {
```

Comment:
It's super heavy to access locks just for metrics. We can maintain a volatile varibles for file size and numSnapshots, and update them once the `registeredKvEntries` is updated. This is an incremental computation and only happens when a new snapshot is created. So it should be more lightweight.